source/tests/universal/common/cases/descriptor/utils.py

```diff
@@ -93,3 +98,133 @@ def test_exclude_types(
                 mapping=mapping_device,
             )
             np.testing.assert_allclose(rd0, rd_ex)
+
+    def test_change_type_map(self):
+        if (
+            not self.module.mixed_types()
+            or getattr(self.module, "sel_no_mixed_types", None) is not None
+        ):
+            # skip if not mixed_types
+            return
+        coord_ext_device = self.coord_ext
+        atype_ext_device = self.atype_ext
+        nlist_device = self.nlist
+        mapping_device = self.mapping
+        # type_map for data and exclude_types
+        original_type_map = ["O", "H"]
+        full_type_map_test = [
+            "H",
+            "He",
+            "Li",
+            "Be",
+            "B",
+            "C",
+            "N",
+            "O",
+            "F",
+            "Ne",
+            "Na",
+            "Mg",
+            "Al",
+            "Si",
+            "P",
+            "S",
+            "Cl",
+            "Ar",
+        ]  # 18 elements
+        for old_tm, new_tm, em, econf in itertools.product(
+            [
+                deepcopy(full_type_map_test),  # 18 elements
+                deepcopy(
+                    full_type_map_test[:16]
+                ),  # 16 elements, double of tebd default first dim
+                deepcopy(full_type_map_test[:8]),  # 8 elements, tebd default first dim
+                ["H", "O"],  # slimmed types
+            ],  # old_type_map
+            [
+                deepcopy(full_type_map_test),  # 18 elements
+                deepcopy(
+                    full_type_map_test[:16]
+                ),  # 16 elements, double of tebd default first dim
+                deepcopy(full_type_map_test[:8]),  # 8 elements, tebd default first dim
+                ["H", "O"],  # slimmed types
+            ],  # new_type_map
+            [[], [[0, 1]], [[1, 1]]],  # exclude_types for original_type_map
+            [False, True],  # use_econf_tebd
+        ):
+            if len(old_tm) >= len(new_tm):
+                continue
```

Comment:
why skipped? please provide the reason

Reply:
Deleted. It's a debug phrase when developing type_map extending and mistakenly kept.